### PR TITLE
Change the image for pull-kueue-build-image-main

### PR DESCRIPTION
**Pull request by tenzen-y**
Signed-off-by: tenzen-y <yuki.iwai.tz@gmail.com>

I changed the image for pull-kueue-build-image-main.

ref: https://github.com/kubernetes-sigs/kueue/pull/379

/assign @alculquicondor

### CODE REVIEW

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml [hunk 1]
```diff
@@ -60,7 +60,7 @@ presubmits:
       preset-dind-enabled: "true"
     spec:
       containers:
-      - image: gcr.io/k8s-staging-test-infra/kubekins-e2e:v20220903-da891f21aa-master
```
Comment (alculquicondor) [inline config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:63]: probably need to set the environment variable too?

Although that's when the presubmit might fail for older releases :(

Comment (tenzen-y) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:63]: > probably need to set the environment variable too?

Does this mean ` DOCKER_BUILDX_CMD`?

Comment (alculquicondor) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:63]: No, it should be `IMAGE_BUILD_CMD`, the one from the `image-build` directive.


Maybe we can add two presubmits, one for old branches (release-0.1 and release-0.2) and one for the rest. This should be possible:

https://github.com/kubernetes/test-infra/blob/1d9942e1a77c183f41c809dc9a767ff220d5fe48/prow/jobs.md#how-to-configure-new-jobs

Comment (tenzen-y) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:63]: Thanks for clarifying.
That makes sense. I agree with you.

Comment (tenzen-y) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:63]: Done.

**Comment by tenzen-y** [comment]
@alculquicondor Since I addressed your comments, please take another look.

**Review by alculquicondor — COMMENTED** [review]
/lgtm
/approve
/label tide/merge-method-squash

File: config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml [hunk 2]
```diff
@@ -58,6 +58,35 @@ presubmits:
...
+    skip_branches:
+    - ^release-0.1$
+    - ^release-0.2$
+    spec:
+      containers:
+      - image: gcr.io/k8s-testimages/gcb-docker-gcloud:v20220830-45cbff55bc
+        securityContext:
+          privileged: true
+        command:
+        - runner.sh
```
[the remark this answers is not in the document]
Comment (tenzen-y) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:70]: I see.  Thanks.
I will create another PR to fix this.

[the remark this answers is not in the document]
Comment (tenzen-y) [reply config/jobs/kubernetes-sigs/kueue/kueue-presubmits.yaml:70]: I have created https://github.com/kubernetes/test-infra/pull/27425.

**Comment by alculquicondor** [comment]
Do you recall why we have different jobs for main and release branches? Can we have just one? Or even none, given that E2E tests also build.

**Comment by tenzen-y** [comment]
> Do you recall why we have different jobs for main and release branches? Can we have just one? Or even none, given that E2E tests also build.

@alculquicondor IIRC, our Dockerfile and build scripts in the release-0.1 and release-0.2 branches don't adapt for multiplatform building. So we must keep having different jobs for the release-0.1 and release-0.2 branches.

Or, we might be able to remove jobs for those branches since we might not create patch releases for those versions.

**Comment by alculquicondor** [comment]
We should have left a comment in the test-infra yaml :sweat_smile: 

> Or, we might be able to remove jobs for those branches since we might not create patch releases for those versions.

We can do so after we release 0.3

**Comment by tenzen-y** [comment]
> We should have left a comment in the test-infra yaml 😅

Yea, this is my bad :_( 

> We can do so after we release 0.3

Agree.

**Pull request merged**